saltproc/abc.py

```diff
@@ -0,0 +1,163 @@
+from abc import ABC, abstractmethod
+
+class Depcode(ABC):
```

Comment:
Couple things:
- are you importing the ABC class from this module into this module?
- Did you name this class `ABC` as in "abstract class"? Is this class ever seen by the user? If it is then this feels like a not-so intuitive class name for a user to have to remember. If the user doesn't see this, then disregard.

Reply:
This pattern indicates that `Depcode` is an abstract class -- i.e. it inherits from the `abc.ABC` class. The purpose of abstract classes is to provide a template for future class creation. The expected usage is something like

```py 
MyClass(Depcode)

def read_depcode_info():
  do stuff
```

where `MyClass` overrides the `read_depcode_info` method (which doesn't currently do anything).

Reply:
I get the `Depcode` being an abstract class part. My confusion is where the `ABC` class is coming from. Is `ABC` is being imported from the `abc` module into an `abc` module? If so, then is it another `abc` module this is being pulled from? Or is the `abc` module it's importing from a python package, like numpy or pandas?

Reply:
He's importing from the latter. My suggestion was to rename the file to avoid this confusion.

Reply:
I support this suggestion. Clearly I was confused by this.

Reply:
I picked `abc.py` as it can hold many different classes that use abstract functions, but perhaps this isn't the best way of doing things.

Reply:
If that's the case, maybe just write out `abstract_classes.py`?